docs/source/json-api/production-setup.rst

```diff
@@ -0,0 +1,300 @@
+the tables again.
+
+If you are not familiar with JDBC URLs, we recommend reading the `PostgreSQL JDBC documentation <https://jdbc.postgresql.org/documentation/head/connect.html>`__
+for more information.
+
+.. note:: The full list of query store configuration flags supported can be seen by running ``daml json-api --help``.
+
+
+Security and privacy
+********************
+
+For an *HTTP-JSON API* server, all data is maintained by the operator of the deployment.
+Thus, it is their responsibility to ensure that the data abides by the necessary
+regulations and confidentiality expectations.
+
+It is recommended to use the tools documented by PostgreSQL to protect data at
+rest and using a secure communication channel between the *HTTP-JSON API* server and the PostgreSQL server.
+
+To protect data in transit and over untrusted networks, the *HTTP-JSON API* server provides
+TLS support, to enable TLS you need to specify the private key for your server and the
+certificate chain via ``daml json-api --pem server.pem --crt server.crt``. You can also
+set a custom root CA certificate used to validate client certificates via ``--cacrt ca.crt``
+
+
+Architecture
+************
+
+Components
+----------
+
+A production setup of the *HTTP-JSON API* will involve the following components:
+
+- the *HTTP-JSON API* server
+- the query store backend database server
+- the ledger
+
+*HTTP-JSON API* server exposes an API to interact with the Ledger and it uses JDBC to interact
+with its underlying query store for caching and serving data efficiently.
+
+The *HTTP-JSON API* server releases are regularly tested with OpenJDK 8 on a 64-bit x86 architecture,
+with Ubuntu 20.04, macOS 11.5.2 and Windows Server 2016.
+
+In production, we recommend running on a 64-bit x86 architecture in a Linux
+environment. This environment should have a Java SE Runtime Environment such
+as OpenJDK JRE and must be compatible with OpenJDK version 1.8.0_202 or later.
+We recommend using PostgreSQL server as query-store, most of our tests have
+been done with servers running version > 10.
+
+
+Scaling and Redundancy
+**********************
+
+.. note:: This section of the document only talks about scaling and redundancy setup for the *HTTP-JSON API* server. In all of the recommendations suggested below we assume that the JSON API always interacts with a single participant on the ledger.
+
+We advise that the *HTTP-JSON API* server and *query store* components to have dedicated
+computation and memory resources available to them. This can be achieved via
+containerization or setting them up on independent physical servers. Ensure that the two
+components are **physically co-located** to reduce network latency for
+communication. The scaling and availability aspects heavily rely on the interactions between
+the core components listed above.
+
+With respect to scaling we recommend to follow the general advice in trying to
```

Comment:
```suggestion
With respect to vertical scaling we recommend to follow the general advice in trying to
```

Otherwise it's a bit weird that we have one paragraph that covers "scaling" and then another one that covers "in general", "horizontal" scaling.